cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -600,7 +601,10 @@ func getUpcomingNodeInfos(registry *clusterstate.ClusterStateRegistry, nodeInfos
 			continue
 		}
 		for i := 0; i < numberOfNodes; i++ {
-			upcomingNodes = append(upcomingNodes, nodeTemplate.Node())
+			// Ensure new nodes having different names because nodeName would used as a map key.
+			node := nodeTemplate.Node().DeepCopy()
+			node.Name = fmt.Sprintf("%s-%d", node.Name, rand.Int63())
```

Comment:
will this affect non-Azure nodes as well?

Reply:
Yep, it would affect non-Azure nodes, but I think it should be fixed for all cloud providers.

@losipiuk @mwielgus Could you help to take a look at this?

Reply:
It feels like serious bug. Thanks for spotting and fixing that.
One note here. Could you please use counter (increased atomically using AddUint64) instead random value. 

cc: @vivekbagade. Vivek could you please scan the code and see if there are no more places where we use node name as dictionary key.

Reply:
We use node name as a dictionary key in a few places. Need to check if they could be causing any issues. Even if they are not, we probably should change this to avoid future issues.

Reply:
This is because of CreateNodeNameToInfoMap func that could potentially mask a few nodes.

Reply:
Thanks for the suggestion, would update the PR. 

> cc: @vivekbagade. Vivek could you please scan the code and see if there are no more places where we use node name as dictionary key.
> We could also add a sanity check in filterOutSchedulableUsingPacking to detect situation when it gets list of nodes with repeating names.
> Also probably we should use UID (not name) as map key.

That looks good. We can do the check and optimization after the code scan.

Reply:
>  Could you please use counter (increased atomically using AddUint64) instead random value.

Actually, the index would be ok. The node name here only used for this single filterOutSchedulableUsingPacking() step.

Reply:
@losipiuk  Updated, PTAL